@xen-orchestra/web/src/components/DonutWithLegends.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+  <div class="donut-with-legends">
+    <DonutChart :segments="donutSegments" :icon="icon" :max-value="maxValue" />
+    <ul class="legends">
+      <UiLegend
+        v-for="(segment, index) in segments"
+        :key="index"
+        :color="segment.color"
+        :value="segment.value"
+        :unit="segment.unit"
+        :tooltip="segment.tooltip"
+      >
+        {{ segment.label }}
+      </UiLegend>
+    </ul>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { DonutSegment } from '@core/types/donut-chart.type'
+import type { LegendColor } from '@core/types/ui-legend.type'
+import DonutChart from '@core/components/DonutChart.vue'
+import UiLegend from '@core/components/UiLegend.vue'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  segments: {
+    label: string
+    value: number
+    unit?: string
+    color: LegendColor
+    tooltip?: string
+  }[]
+  icon: IconDefinition
```

Comment:
Since the `icon` prop is optional in `DonutChart`, it should be optional as well:
```suggestion
  icon?: IconDefinition
```